src/components/dashboard/documents-tab.js

```diff
@@ -0,0 +1,144 @@
+import PropTypes from 'prop-types';
+import routes from '../../utils/routes.js';
+import FilterInput from '../filter-input.js';
+import { useUser } from '../user-context.js';
+import DocumentCard from '../document-card.js';
+import { useTranslation } from 'react-i18next';
+import React, { useEffect, useState } from 'react';
+import { Button, Checkbox, Pagination, Spin } from 'antd';
+import { DOC_VIEW_QUERY_PARAM } from '../../domain/constants.js';
+import DocumentMetadataModal from '../document-metadata-modal.js';
+import { contributedDocumentMetadataShape } from '../../ui/default-prop-types.js';
+import { DOCUMENT_METADATA_MODAL_MODE } from '../document-metadata-modal-utils.js';
+
+const PAGE_SIZE = 8;
+const DEFAULT_PAGE_NUMBER = 1;
+
+function DocumentsTab({ documents, loading }) {
+  const user = useUser();
+  const { t } = useTranslation('documentsTab');
+
+  const [filterText, setFilterText] = useState(null);
+  const [pageDocuments, setPageDocuments] = useState([]);
+  const [ownDocumentsOnly, setOwnDocumentsOnly] = useState(false);
+  const [displayedDocuments, setDisplayedDocuments] = useState([]);
+  const [currentPage, setCurrentPage] = useState(DEFAULT_PAGE_NUMBER);
+
+  useEffect(() => {
+    setCurrentPage(DEFAULT_PAGE_NUMBER);
+  }, [documents, filterText, ownDocumentsOnly]);
+
+  useEffect(() => {
+    const lowerCasedFilter = (filterText || '').toLowerCase();
+    let filteredDocuments = filterText
+      ? documents.filter(doc => {
+        return doc.title.toLowerCase().includes(lowerCasedFilter)
+            || doc.description.toLowerCase().includes(lowerCasedFilter)
+            || doc.createdBy.displayName.toLowerCase().includes(lowerCasedFilter)
+            || doc.updatedBy.displayName.toLowerCase().includes(lowerCasedFilter);
+      })
+      : documents;
+
+    filteredDocuments = ownDocumentsOnly
+      ? filteredDocuments.filter(doc => doc.createdBy._id === user._id)
+      : filteredDocuments;
+
+    const indexOfFirstPageDocument = (currentPage - 1) * PAGE_SIZE;
```

Comment:
Actually this never occurs because we have a separate reset of the current page:

```
  useEffect(() => {
    setCurrentPage(DEFAULT_PAGE_NUMBER);
  }, [documents, filterText, ownDocumentsOnly]);
```

Reply:
Ah, true. OK, withdrawn :-)